sonar-dotnet-shared-library/src/main/java/org/sonarsource/dotnet/shared/plugins/DotNetSensor.java

```diff
@@ -73,16 +72,24 @@ public void execute(SensorContext context) {
   }
 
   private boolean shouldExecuteOnProject(FileSystem fs) {
-    if (!filesToAnalyze(fs).iterator().hasNext()) {
-      LOG.debug("No files to analyze. Skip Sensor.");
+    boolean hasMainFiles = hasFilesOfType(fs, Type.MAIN);
+    boolean hasTestFiles = hasFilesOfType(fs, Type.TEST);
+
+    if (!hasMainFiles) {
+      if (hasTestFiles) {
+        String noResultsMessage = "Your SonarQube/SonarCloud project will not have results for " + pluginMetadata.languageKey() + " files.";
+        LOG.warn("This solution contains only TEST files. Because there are no MAIN files, this sensor will be skipped. " + noResultsMessage);
+      } else {
+        LOG.debug("No files to analyze. Skip Sensor.");
```

Comment:
given that this is DEBUG, I think I can add `Your SonarQube/SonarCloud project will not have results for LANG files.` here too, right?

Verbose logs should make things obvious.

Reply:
I think it's good as it is. The "No files to analyze" is clear enough. Given it's a DEBUG.

I would not scare or mislead users using S4MSB for SQL and other projects.